cmd/api/src/services/graphify/decoders.go

```diff
@@ -41,7 +40,7 @@ func decodeBasicData[T any](batch *TimestampedBatch, decoder *json.Decoder, conv
 	var (
 		count         = 0
 		convertedData ConvertedData
-		errs          = util.NewErrorCollector()
+		errs          = newGraphifyErrorBuilder()
```

Comment:
This feels like a good opportunity to just use `[]error`, append errors to `errs`, and then `errors.Join()` at the end to get a singular error value (containing the others). This not only is simpler and doesn't require the new file, but also preserves errors.Is and errors.As semantics if we ever need them in higher code. (`errors.Join` on an empty list or even a list of nils is always `nil` so you don't need to do any extra checks either)